pkg/dashboard/resource/apigateway.go

```diff
@@ -48,14 +49,21 @@ func (agr *apiGatewayResource) GetAll(request *http.Request) (map[string]restful
 	}
 
 	exportFunction := agr.GetURLParamBoolOrDefault(request, restful.ParamExport, false)
+	projectName := request.Header.Get("x-nuclio-project-name")
 
-	return agr.GetAllByNamespace(namespace, exportFunction)
+	return agr.GetAllByNamespace(namespace, projectName, exportFunction)
 }
 
 // GetAll returns all api-gateways
-func (agr *apiGatewayResource) GetAllByNamespace(namespace string, exportFunction bool) (map[string]restful.Attributes, error) {
+func (agr *apiGatewayResource) GetAllByNamespace(namespace string, projectName string, exportFunction bool) (map[string]restful.Attributes, error) {
```

Comment:
is there a reason it is public? (if it consumed by other, please ensure function comment above is fixed `GetAll` vs `GetAllByNamespace` )

now that you added project name, you may need to change function signature name as well

Reply:
to make this function easier to expand you may want to simply pass a `GetAPIGatewaysOptions` structure that contain all parameters to query with to get resources

Reply:
GetAllByNamespace is used by project resource to export api gateways - that's why it's public